projects/common/src/utilities/formatters/string/highlight.pipe.ts

```diff
@@ -1,20 +1,41 @@
 import { Pipe, PipeTransform } from '@angular/core';
+import { isArray } from 'lodash-es';
 import { assertUnreachable } from '../../lang/lang-utils';
 
 @Pipe({ name: 'htHighlight' })
 export class HighlightPipe implements PipeTransform {
   public transform(
     fullText: string,
-    highlightedText: string | undefined,
+    highlightSnippets: string | undefined | TextHighlightConfig[],
     highlightType: HighlightType = 'mark'
   ): string {
+    if (isArray(highlightSnippets)) {
```

Comment:
nit - rather than having a separate path, for array vs string, normalize the inputs first and then put everything through the same path. Also, this is a textbook use case for `reduce`  😉